test/utils/nrosched/nrosched.go

```diff
@@ -93,42 +93,29 @@ func CheckPODWasScheduledWith(k8sCli *kubernetes.Clientset, podNamespace, podNam
 	return checkPODEvents(k8sCli, podNamespace, podName, schedulerName, isScheduledWith)
 }
 
-func CheckNROSchedulerAvailable(cli client.Client, nroSchedName string) *nropv1alpha1.NUMAResourcesScheduler {
-	nroSchedObj := &nropv1alpha1.NUMAResourcesScheduler{}
-	Eventually(func() bool {
-		By(fmt.Sprintf("checking %q for the condition Available=true", nroSchedName))
-
-		err := cli.Get(context.TODO(), client.ObjectKey{Name: NROSchedObjectName}, nroSchedObj)
-		if err != nil {
-			klog.Warningf("failed to get the scheduler resource: %v", err)
-			return false
-		}
-
-		cond := status.FindCondition(nroSchedObj.Status.Conditions, status.ConditionAvailable)
-		if cond == nil {
-			klog.Warningf("missing conditions in %v", nroSchedObj)
-			return false
-		}
-
-		klog.Infof("condition: %v", cond)
-
-		return cond.Status == metav1.ConditionTrue
-	}, 5*time.Minute, 10*time.Second).Should(BeTrue(), "Scheduler condition did not become available")
-	return nroSchedObj
+func CheckNROSchedulerAvailable(cli client.Client) *nropv1alpha1.NUMAResourcesScheduler {
+	return isSchedulerAvailable(cli)
 }
 
 func IsSchedulingErrorCannotAlign(msg string) bool {
 	return strings.Contains(msg, ErrorCannotAlignPod)
 }
 
 func GetNROSchedulerName(cli client.Client) string {
+	return isSchedulerAvailable(cli).Spec.SchedulerName
+}
+
+func isSchedulerAvailable(cli client.Client) *nropv1alpha1.NUMAResourcesScheduler {
```

Comment:
but this is not telling if a scheduler (which one?) is available, it is getting the scheduler object data. So the name doesn't seem really fitting.

Reply:
Yeah, this certainly needs a change. My initial intention was to return a boolean corresponding to a check if scheduler is available but later decided to encapsulate the common code between CheckNROSchedulerAvailable and GetNROSchedulerName. Anyway, will fix it.